src/Datadog.Trace/Logging/DatadogLogging.cs

```diff
@@ -133,7 +143,20 @@ internal static void SetLogLevel(LogEventLevel logLevel)
 
         internal static void UseDefaultLevel()
         {
-            SetLogLevel(LogEventLevel.Information);
+            SetLogLevel(DefaultLogLevel);
+        }
+
+        private static int GetRateLimit()
+        {
+            string rawRateLimit = EnvironmentHelpers.GetEnvironmentVariable(ConfigurationKeys.LogRateLimit);
+            if (!string.IsNullOrEmpty(rawRateLimit)
+                && int.TryParse(rawRateLimit, out var rate)
+                && (rate < 0))
```

Comment:
Shouldn't that be `rate >= 0` ? That would be nice to have a unit test or two for this

Reply:
Coming back from the future. The RateLimiter refuses rates lower than 1, so it should actually be `rate >= 1`

Reply:
🤦‍♂️

Reply:
Writing unit tests for this class is not easy, given it's all statically initialized from env variables. I could extract this all out to a factory class so we can test things more easily if you think that's worth it? (same goes for testing the log directory creation etc)

Reply:
Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/1153/commits/05de3984613bd001f05f2c5f22b34db3ccb4d2e5, but requires more refactoring to make `DatadogLogging` testable

Reply:
> requires more refactoring to make DatadogLogging testable

Let's add this to the list of testing improvements.

Reply:
> Let's add this to the list of testing improvements.

Done